projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -216,22 +219,24 @@ private InboundFrame(FrameType type, int channel, ReadOnlyMemory<byte> payload,
             _rentedArray = rentedArray;
         }
 
-        private static void ProcessProtocolHeader(Stream reader, ReadOnlySpan<byte> frameHeader)
+        private static void ProcessProtocolHeader(ReadOnlySequence<byte> buffer)
         {
             try
             {
-                if (frameHeader[0] != 'M' || frameHeader[1] != 'Q' || frameHeader[2] != 'P')
+                if (buffer.Length < 8)
                 {
-                    throw new MalformedFrameException("Invalid AMQP protocol header from server");
+                    throw new EndOfStreamException();
                 }
 
-                int serverMinor = reader.ReadByte();
-                if (serverMinor == -1)
+                Span<byte> tempSpan = stackalloc byte[8];
+                buffer.Slice(0, 8).CopyTo(tempSpan);
+
+                if (tempSpan[1] != 'M' || tempSpan[2] != 'Q' || tempSpan[3] != 'P')
                 {
-                    throw new EndOfStreamException();
+                    throw new MalformedFrameException("Invalid AMQP protocol header from server");
                 }
 
-                throw new PacketNotRecognizedException(frameHeader[3], frameHeader[4], frameHeader[5], serverMinor);
+                throw new PacketNotRecognizedException(tempSpan[4], tempSpan[5], tempSpan[6], tempSpan[7]);
```

Comment:
Also why would the copy be necessary here? The values should be copied to the ints in the exception anyway shouldn't they? See https://dotnetfiddle.net/XAsUAK

Reply:
Very cool way to demonstrate that!